integration-tests/src/upgrade/builder.rs

```diff
@@ -0,0 +1,165 @@
+    delegate_to: (Address, Bytes),
+    /// Timestamp after which upgrade can be executed
+    /// If not provided, default value will be used (e.g. upgrade whenever)
+    timestamp: U256,
+    // I don't need more parameters here for now, but in the future if you want
+    // to extend this builder, feel free to do it.
+}
+
+impl ProtocolUpgradeBuilder {
+    /// Create a new `ProtocolUpgradeBuilder` with default values.
+    pub(super) fn new(current_protocol_version: U256, delegate_to: (Address, Bytes)) -> Self {
+        Self {
+            protocol_version: current_protocol_version,
+            force_deployments: None,
+            delegate_to,
+            timestamp: U256::ZERO,
+        }
+    }
+
+    /// Sets the delegation target for L2 upgrade logic.
+    /// Right now, MUST correspond to a real deployed account with bytecode.
+    /// At the time of writing, the upgrade logic MUST be executed during the upgrade,
+    /// so this field is mandatory, even if we don't need to execute any specific logic.
+    /// By default, `UpgradeTester::protocol_upgrade_builder` will deploy a no-op contract
+    /// and set it as the delegate, so this method is only needed if you want to
+    /// execute some specific logic during the upgrade.
+    pub fn delegate_to(mut self, delegate_address: Address, calldata: Bytes) -> Self {
+        self.delegate_to = (delegate_address, calldata);
+        self
+    }
+
+    /// Bump the minor protocol version by the specified amount.
+    pub fn bump_minor(mut self, by: u64) -> Self {
+        let semver_minor_version_multiplier = U256::from(4294967296u64); // 2^32
+        self.protocol_version += semver_minor_version_multiplier * U256::from(by);
+        self
+    }
+
+    /// Bump the patch protocol version by the specified amount.
+    pub fn bump_patch(mut self, by: u64) -> Self {
+        self.protocol_version += U256::from(by);
+        self
+    }
+
+    /// Sets the protocol version to the specified value.
+    pub fn set_version(mut self, version: U256) -> Self {
+        self.protocol_version = version;
+        self
+    }
+
+    /// Optional. Sets the list of contracts to be force-deployed during the upgrade.
+    pub fn with_force_deployments(mut self, deployments: BTreeMap<Address, Bytes>) -> Self {
+        self.force_deployments = Some(deployments);
+        self
+    }
+
+    /// Sets the timestamp after which the upgrade can be executed.
+    pub fn with_timestamp(mut self, timestamp: U256) -> Self {
+        self.timestamp = timestamp;
+        self
+    }
+
+    /// Builds the `ProposedUpgrade` struct.
+    pub fn build(self) -> ProposedUpgrade {
+        let (delegate_to_address, delegate_to_calldata) = self.delegate_to;
+
+        const DEPLOYER_ADDRESS: &str = "0x0000000000000000000000000000000000008007";
+        const COMPLEX_UPGRADER_ADDRESS: &str = "0x000000000000000000000000000000000000800f";
+
+        const REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT: u32 = 800;
```

Comment:
There is `zksync_os_types::REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_BYTE` already that represents the same thing